var/spack/repos/builtin/packages/charliecloud/package.py

```diff
@@ -13,6 +13,19 @@ class Charliecloud(MakefilePackage):
     url      = "https://github.com/hpc/charliecloud/releases/download/v0.9.10/charliecloud-0.9.10.tar.gz"
     git      = "https://github.com/hpc/charliecloud.git"
 
+    depends_on('python@3.4:', type=('build', 'run'))
+
+    # experimental builder (ch-grow)
+    variant('ch-grow', default=False, description='Bundle dependencies for unprivileged builder')
+    depends_on('py-lark-parser', type='run', when='+ch-grow')
+
+    # man pages and html docs
+    variant('docs', default=False, description='Build man pages and html docs')
+    depends_on('rsync',               type='build', when='+doc')
+    depends_on('py-sphinx',           type='build', when='+doc')
+    depends_on('py-sphinx-rtd-theme', type='build', when='+doc')
```

Comment:
I would move all the variants and dependencies below the versions, since the versions are more likely to be updated from time to time.